pkg/executor/sortexec/sort.go

```diff
@@ -52,42 +54,74 @@ type SortExec struct {
 	memTracker  *memory.Tracker
 	diskTracker *disk.Tracker
 
-	// sortPartitions is the chunks to store row values for partitions. Every partition is a sorted list.
-	sortPartitions []*sortPartition
+	IsUnparallel bool
 
-	// multiWayMerge uses multi-way merge for spill disk.
-	// The multi-way merge algorithm can refer to https://en.wikipedia.org/wiki/K-way_merge_algorithm
-	multiWayMerge *multiWayMerge
-	// spillAction save the Action for spill disk.
-	spillAction *sortPartitionSpillDiskAction
+	finishCh chan struct{}
+
+	Unparallel struct {
+		Idx int
+
+		// sortPartitions is the chunks to store row values for partitions. Every partition is a sorted list.
+		sortPartitions []*sortPartition
+
+		// multiWayMerge uses multi-way merge for spill disk.
+		// The multi-way merge algorithm can refer to https://en.wikipedia.org/wiki/K-way_merge_algorithm
+		multiWayMerge *multiWayMergeImpl
+
+		// spillAction save the Action for spill disk.
+		spillAction *sortPartitionSpillDiskAction
+	}
+
+	Parallel struct {
+		chunkChannel chan *chunk.Chunk
+		workers      []*parallelSortWorker
+
+		// Each worker will put their results into the given iter
+		sortedRowsIters []*chunk.Iterator4Slice
+		merger          *multiWayMerger
+		mergerLock      *sync.Mutex
+
+		resultChannel chan rowWithError
+	}
 
 	enableTmpStorageOnOOM bool
 }
 
 // Close implements the Executor Close interface.
 func (e *SortExec) Close() error {
-	for _, partition := range e.sortPartitions {
-		err := partition.close()
-		if err != nil {
-			return err
+	close(e.finishCh)
+	if e.Unparallel.spillAction != nil {
+		e.Unparallel.spillAction.SetFinished()
+	}
+
+	if e.IsUnparallel {
+		for _, partition := range e.Unparallel.sortPartitions {
+			partition.close()
 		}
+	} else {
+		for {
+			_, ok := <-e.Parallel.chunkChannel
```

Comment:
What happend when calling `Close` after `Open` without calling `Next`?

Reply:
> What happend when calling `Close` after `Open` without calling `Next`?

I will fix it.